Get mega-melt working on CI

For a brief time, I enabled `bin/tests.sh` (i.e. the "mega-melt") on Travis CI. Unfortunately, there were problems.

Here is an example Travis build ([build 467](https://travis-ci.org/github/scijava/pom-scijava/builds/712369079)):

<details><summary>Relevant portion of the build log</summary>

```
Generating mega-melt project... Done!
Validating mega-melt project... Done!
Generating melting pot...
[INFO] Local Maven project: /home/travis/build/scijava/pom-scijava/target/mega-melt
[INFO] Determining project dependencies
[INFO] Processing project dependencies
[INFO] Sholl_Analysis: fetching component source
[INFO] jvrpn: fetching component source
[WARNING] graphics.scenery:jvrpn:1.1.0: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: v1.0.0^{}: bad variable name
[ERROR] Could not fetch project source for graphics.scenery:jvrpn:1.1.0
[INFO] spirvcrossj: fetching component source
[WARNING] graphics.scenery:spirvcrossj:0.6.0-1.1.106.0: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: spirvcrossj-0.2.0^{}: bad variable name
[ERROR] Could not fetch project source for graphics.scenery:spirvcrossj:0.6.0-1.1.106.0
[INFO] scifio-bf-compat: fetching component source
[INFO] scifio-cli: fetching component source
[INFO] scifio-hdf5: fetching component source
[INFO] scifio-jai-imageio: fetching component source
[INFO] scifio: fetching component source
[INFO] scifio-lifesci: fetching component source
[INFO] scifio-ome-xml: fetching component source
[INFO] jitk-tps: fetching component source
[INFO] mpicbg_: fetching component source
[INFO] mpicbg: fetching component source
[INFO] ij1-patcher: fetching component source
[INFO] ij: fetching component source
[WARNING] net.imagej:ij:1.52p: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: v1.38u: bad variable name
[ERROR] Could not fetch project source for net.imagej:ij:1.52p
[INFO] imagej-common: fetching component source
[INFO] imagej-deprecated: fetching component source
[INFO] imagej: fetching component source
[INFO] imagej-launcher: fetching component source
[INFO] imagej-legacy: fetching component source
[INFO] imagej-matlab: fetching component source
[INFO] imagej-mesh-io: fetching component source
[INFO] imagej-mesh: fetching component source
[INFO] imagej-notebook: fetching component source
[INFO] imagej-opencv: fetching component source
[INFO] imagej-ops: fetching component source
[INFO] imagej-plugins-batch: fetching component source
[INFO] imagej-plugins-commands: fetching component source
[INFO] imagej-plugins-tools: fetching component source
[INFO] imagej-plugins-uploader-ssh: fetching component source
[INFO] imagej-plugins-uploader-webdav: fetching component source
[INFO] imagej-scripting: fetching component source
[INFO] imagej-tensorflow: fetching component source
[INFO] imagej-ui-awt: fetching component source
[INFO] imagej-ui-swing: fetching component source
[INFO] imagej-updater: fetching component source
[INFO] op-finder: fetching component source
[INFO] imglib2-algorithm-fft: fetching component source
[INFO] imglib2-algorithm-gpl: fetching component source
[INFO] imglib2-algorithm: fetching component source
[INFO] imglib2-cache: fetching component source
[INFO] imglib2-ij: fetching component source
[INFO] imglib2: fetching component source
[INFO] imglib2-label-multisets: fetching component source
[WARNING] net.imglib2:imglib2-label-multisets:0.9.0: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: 0.1.0^{}: bad variable name
[ERROR] Could not fetch project source for net.imglib2:imglib2-label-multisets:0.9.0
[INFO] imglib2-realtransform: fetching component source
[INFO] imglib2-roi: fetching component source
[INFO] imglib2-script: fetching component source
[INFO] imglib2-ui: fetching component source
[INFO] imglib2-unsafe: fetching component source
[INFO] BigStitcher: fetching component source
[INFO] multiview-reconstruction: fetching component source
[INFO] multiview-simulation: fetching component source
[INFO] bonej-legacy-plugins_: fetching component source
[INFO] bonej-legacy-util_: fetching component source
[INFO] bonej-ops: fetching component source
[INFO] bonej-plugins: fetching component source
[INFO] bonej-utilities: fetching component source
[INFO] H5J_Loader_Plugin: fetching component source
[INFO] n5-aws-s3: fetching component source
[INFO] n5-google-cloud: fetching component source
[INFO] n5-hdf5: fetching component source
[INFO] n5-imglib2: fetching component source
[INFO] n5: fetching component source
[INFO] n5-viewer_fiji: fetching component source
[WARNING] org.janelia.saalfeldlab:n5-viewer_fiji:2.2.0: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: 1.1.0: bad variable name
[ERROR] Could not fetch project source for org.janelia.saalfeldlab:n5-viewer_fiji:2.2.0
[INFO] batch-processor: fetching component source
[INFO] minimaven: fetching component source
[INFO] native-lib-loader: fetching component source
[INFO] parsington: fetching component source
[INFO] scijava-cache: fetching component source
[INFO] scijava-common: fetching component source
[INFO] scijava-config: fetching component source
[INFO] scijava-grab: fetching component source
[INFO] scijava-io-http: fetching component source
[INFO] scijava-java3d: fetching component source
[INFO] scijava-listeners: fetching component source
[INFO] scijava-log-slf4j: fetching component source
[INFO] scijava-optional: fetching component source
[INFO] scijava-plugins-commands: fetching component source
[INFO] scijava-plugins-io-table: fetching component source
[INFO] scijava-plugins-platforms: fetching component source
[INFO] scijava-plugins-text-markdown: fetching component source
[INFO] scijava-plugins-text-plain: fetching component source
[INFO] scijava-search: fetching component source
[INFO] scijava-table: fetching component source
[INFO] scijava-ui-awt: fetching component source
[INFO] scijava-ui-swing: fetching component source
[INFO] script-editor: fetching component source
[INFO] scripting-beanshell: fetching component source
[INFO] scripting-clojure: fetching component source
[INFO] scripting-groovy: fetching component source
[INFO] scripting-java: fetching component source
[INFO] scripting-javascript: fetching component source
[INFO] scripting-jruby: fetching component source
[INFO] scripting-jython: fetching component source
[INFO] scripting-kotlin: fetching component source
[INFO] scripting-matlab: fetching component source
[INFO] scripting-renjin: fetching component source
[INFO] scripting-scala: fetching component source
[INFO] swing-checkbox-tree: fetching component source
[INFO] ui-behaviour: fetching component source
[INFO] 3D_Blob_Segmentation: fetching component source
[INFO] 3D_Objects_Counter: fetching component source
[INFO] 3D_Viewer: fetching component source
[INFO] AnalyzeSkeleton_: fetching component source
[INFO] Anisotropic_Diffusion_2D: fetching component source
[INFO] Archipelago_Plugins: fetching component source
[INFO] Arrow_: fetching component source
[INFO] Auto_Local_Threshold: fetching component source
[INFO] Auto_Threshold: fetching component source
[INFO] BalloonSegmentation_: fetching component source
[INFO] bigdataviewer-core: fetching component source
[INFO] bigdataviewer_fiji: fetching component source
[INFO] bigdataviewer-server: fetching component source
[INFO] bigdataviewer-vistools: fetching component source
[INFO] bigvolumeviewer: fetching component source
[INFO] bigwarp_fiji: fetching component source
[INFO] bij: fetching component source
[INFO] blockmatching_: fetching component source
[INFO] Bug_Submitter: fetching component source
[INFO] bUnwarpJ_: fetching component source
[INFO] Calculator_Plus: fetching component source
[INFO] Cell_Counter: fetching component source
[INFO] Colocalisation_Analysis: fetching component source
[INFO] Color_Histogram: fetching component source
[INFO] Color_Inspector_3D: fetching component source
[WARNING] sc.fiji:Color_Inspector_3D:2.5.0: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: Color_Inspector_3D-2.3.1^{}: bad variable name
[ERROR] Could not fetch project source for sc.fiji:Color_Inspector_3D:2.5.0
[INFO] Colour_Deconvolution: fetching component source
[INFO] Correct_3D_Drift: fetching component source
[INFO] CorrectBleach_: fetching component source
[INFO] CPU_Meter: fetching component source
[INFO] Descriptor_based_registration: fetching component source
[INFO] Dichromacy_: fetching component source
[INFO] Directionality_: fetching component source
[INFO] Feature_Detection: fetching component source
[INFO] Fiji_Archipelago: fetching component source
[INFO] Fiji_Developer: fetching component source
[INFO] fiji: fetching component source
[INFO] fiji-lib: fetching component source
[INFO] Fiji_Package_Maker: fetching component source
[INFO] Fiji_Plugins: fetching component source
[INFO] FlowJ_: fetching component source
[INFO] FS_Align_TrakEM2: fetching component source
[INFO] Graph_Cut: fetching component source
[INFO] Gray_Morphology: fetching component source
[INFO] HDF5_Vibez: fetching component source
[INFO] Helmholtz_Analysis: fetching component source
[INFO] IJ_Robot: fetching component source
[INFO] Image_5D: fetching component source
[INFO] Image_Expression_Parser: fetching component source
[INFO] Interactive_3D_Surface_Plot: fetching component source
[INFO] IO_: fetching component source
[INFO] IsoData_Classifier: fetching component source
[INFO] Kuwahara_Filter: fetching component source
[INFO] KymographBuilder: fetching component source
[INFO] Lasso_and_Blow_Tool: fetching component source
[INFO] legacy-imglib1: fetching component source
[INFO] level_sets: fetching component source
[INFO] Linear_Kuwahara: fetching component source
[INFO] LocalThickness_: fetching component source
[INFO] LSM_Reader: fetching component source
[INFO] LSM_Toolbox: fetching component source
[INFO] Manual_Tracking: fetching component source
[INFO] M_I_P: fetching component source
[INFO] MTrack2_: fetching component source
[INFO] Multi_Kymograph: fetching component source
[INFO] OMEVisual: fetching component source
[INFO] pal-optimization: fetching component source
[INFO] panorama_: fetching component source
[INFO] PIV_analyser: fetching component source
[INFO] QuickPALM_: fetching component source
[INFO] RATS_: fetching component source
[INFO] Reconstruct_Reader: fetching component source
[INFO] register_virtual_stack_slices: fetching component source
[INFO] registration_3d: fetching component source
[INFO] Samples_: fetching component source
[INFO] Series_Labeler: fetching component source
[INFO] Simple_Neurite_Tracer: fetching component source
[INFO] Siox_Segmentation: fetching component source
[INFO] Skeletonize3D_: fetching component source
[INFO] spim_data: fetching component source
[INFO] SPIM_Opener: fetching component source
[INFO] SPIM_Registration: fetching component source
[INFO] SplineDeformationGenerator_: fetching component source
[INFO] Stack_Manipulation: fetching component source
[INFO] Statistical_Region_Merging: fetching component source
[INFO] Stitching_: fetching component source
[INFO] Sync_Win: fetching component source
[INFO] T2-NIT: fetching component source
[INFO] T2-TreelineGraph: fetching component source
[INFO] Thread_Killer: fetching component source
[INFO] Time_Lapse: fetching component source
[INFO] Time_Stamper: fetching component source
[INFO] ToAST_: fetching component source
[INFO] TopoJ_: fetching component source
[INFO] TrackMate_: fetching component source
[INFO] Trainable_Segmentation: fetching component source
[WARNING] sc.fiji:Trainable_Segmentation:3.2.34: improper scm tag value; scanning remote tags...
tests/../target/mega-melt/melting-pot.sh: 428: local: Trainable_Segmentation-2.1.1: bad variable name
[ERROR] Could not fetch project source for sc.fiji:Trainable_Segmentation:3.2.34
[INFO] TrakEM2_Archipelago: fetching component source
[INFO] TrakEM2_: fetching component source
[INFO] trakem2_tps: fetching component source
[INFO] trakem2-transform: fetching component source
[INFO] Vaa3d_Reader: fetching component source
[INFO] Vaa3d_Writer: fetching component source
[INFO] VectorString: fetching component source
[INFO] VIB_: fetching component source
[INFO] VIB-lib: fetching component source
[INFO] Video_Editing: fetching component source
[INFO] View5D_: fetching component source
[INFO] Volume_Calculator: fetching component source
[INFO] Volume_Viewer: fetching component source
[INFO] weave_jy2java: fetching component source
[INFO] z_spacing: fetching component source
[INFO] sciview: fetching component source
[INFO] Processing changed components
[INFO] Generating build scripts
[INFO] Skipping the build; run melt.sh to do it.
[INFO] Melt complete: /home/travis/build/scijava/pom-scijava/target/mega-melt
Adjusting melting pot build script... Done!
Melting the pot...
[SUCCESS] LOCAL/PROJECT
[SUCCESS] ca.mcgill/Sholl_Analysis
[1 minutes elapsed]
[SUCCESS] io.scif/scifio
[SUCCESS] io.scif/scifio-bf-compat
[SUCCESS] io.scif/scifio-cli
[SUCCESS] io.scif/scifio-hdf5
[2 minutes elapsed]
[SUCCESS] io.scif/scifio-jai-imageio
[SUCCESS] io.scif/scifio-lifesci
[SUCCESS] io.scif/scifio-ome-xml
[SUCCESS] jitk/jitk-tps
[SUCCESS] mpicbg/mpicbg/mpicbg
[SUCCESS] mpicbg/mpicbg_/mpicbg_
[3 minutes elapsed]
[SUCCESS] net.imagej/ij1-patcher
[SUCCESS] net.imagej/imagej
[4 minutes elapsed]
[SUCCESS] net.imagej/imagej-common
[SUCCESS] net.imagej/imagej-deprecated
[SUCCESS] net.imagej/imagej-launcher
[SUCCESS] net.imagej/imagej-legacy
[SUCCESS] net.imagej/imagej-matlab
[5 minutes elapsed]
[SUCCESS] net.imagej/imagej-mesh
[SUCCESS] net.imagej/imagej-mesh-io
[SUCCESS] net.imagej/imagej-notebook
[SUCCESS] net.imagej/imagej-opencv
[6 minutes elapsed]
[7 minutes elapsed]
[8 minutes elapsed]
[9 minutes elapsed]
[10 minutes elapsed]
[11 minutes elapsed]
[SUCCESS] net.imagej/imagej-ops
[SUCCESS] net.imagej/imagej-plugins-batch
[SUCCESS] net.imagej/imagej-plugins-commands
[SUCCESS] net.imagej/imagej-plugins-tools
[12 minutes elapsed]
[SUCCESS] net.imagej/imagej-plugins-uploader-ssh
[SUCCESS] net.imagej/imagej-plugins-uploader-webdav
[SUCCESS] net.imagej/imagej-scripting
[13 minutes elapsed]
[SUCCESS] net.imagej/imagej-tensorflow
[SUCCESS] net.imagej/imagej-ui-awt
[FAILURE] net.imagej/imagej-ui-swing
[SUCCESS] net.imagej/imagej-updater
[SUCCESS] net.imagej/op-finder
[14 minutes elapsed]
[SUCCESS] net.imglib2/imglib2
[SUCCESS] net.imglib2/imglib2-algorithm
[15 minutes elapsed]
[SUCCESS] net.imglib2/imglib2-algorithm-fft
[SUCCESS] net.imglib2/imglib2-algorithm-gpl
[SUCCESS] net.imglib2/imglib2-cache
[16 minutes elapsed]
[SUCCESS] net.imglib2/imglib2-ij
[SUCCESS] net.imglib2/imglib2-realtransform
[SUCCESS] net.imglib2/imglib2-roi
[17 minutes elapsed]
[SUCCESS] net.imglib2/imglib2-script
[SUCCESS] net.imglib2/imglib2-ui
[SUCCESS] net.imglib2/imglib2-unsafe
[SUCCESS] net.preibisch/BigStitcher
[FAILURE] net.preibisch/multiview-reconstruction
[SUCCESS] net.preibisch/multiview-simulation
[18 minutes elapsed]
[SUCCESS] org.janelia.saalfeldlab/n5
[SUCCESS] org.janelia.saalfeldlab/n5-aws-s3
[SUCCESS] org.janelia.saalfeldlab/n5-google-cloud
[19 minutes elapsed]
[SUCCESS] org.janelia.saalfeldlab/n5-hdf5
[SUCCESS] org.janelia.saalfeldlab/n5-imglib2
[SUCCESS] org.janelia/H5J_Loader_Plugin
[SUCCESS] org.scijava/batch-processor
[SUCCESS] org.scijava/minimaven
[SUCCESS] org.scijava/native-lib-loader
[SUCCESS] org.scijava/parsington
[SUCCESS] org.scijava/scijava-cache
[20 minutes elapsed]
[SUCCESS] org.scijava/scijava-common
[SUCCESS] org.scijava/scijava-config
[SUCCESS] org.scijava/scijava-grab
[SUCCESS] org.scijava/scijava-io-http
[SUCCESS] org.scijava/scijava-java3d
[SUCCESS] org.scijava/scijava-listeners
[SUCCESS] org.scijava/scijava-log-slf4j
[21 minutes elapsed]
[SUCCESS] org.scijava/scijava-optional
[SUCCESS] org.scijava/scijava-plugins-commands
[SUCCESS] org.scijava/scijava-plugins-io-table
[SUCCESS] org.scijava/scijava-plugins-platforms
[SUCCESS] org.scijava/scijava-plugins-text-markdown
[SUCCESS] org.scijava/scijava-plugins-text-plain
[SUCCESS] org.scijava/scijava-search
[SUCCESS] org.scijava/scijava-table
[SUCCESS] org.scijava/scijava-ui-awt
[22 minutes elapsed]
[SUCCESS] org.scijava/scijava-ui-swing
[SUCCESS] org.scijava/script-editor
[SUCCESS] org.scijava/scripting-beanshell
[SUCCESS] org.scijava/scripting-clojure
[SUCCESS] org.scijava/scripting-groovy
[SUCCESS] org.scijava/scripting-java
[SUCCESS] org.scijava/scripting-javascript
[23 minutes elapsed]
[SUCCESS] org.scijava/scripting-jruby
[SUCCESS] org.scijava/scripting-jython
[SUCCESS] org.scijava/scripting-kotlin
[SUCCESS] org.scijava/scripting-matlab
[SUCCESS] org.scijava/scripting-renjin
[24 minutes elapsed]
[SUCCESS] org.scijava/scripting-scala
[SUCCESS] org.scijava/swing-checkbox-tree
[SUCCESS] org.scijava/ui-behaviour
[SUCCESS] sc.fiji/3D_Blob_Segmentation
[SUCCESS] sc.fiji/3D_Objects_Counter
[SUCCESS] sc.fiji/3D_Viewer
[SUCCESS] sc.fiji/AnalyzeSkeleton_
[SUCCESS] sc.fiji/Anisotropic_Diffusion_2D
[25 minutes elapsed]
[SUCCESS] sc.fiji/Archipelago_Plugins
[SUCCESS] sc.fiji/Arrow_
[SUCCESS] sc.fiji/Auto_Local_Threshold
[SUCCESS] sc.fiji/Auto_Threshold
[SUCCESS] sc.fiji/BalloonSegmentation_
[SUCCESS] sc.fiji/Bug_Submitter
[SUCCESS] sc.fiji/CPU_Meter
[SUCCESS] sc.fiji/Calculator_Plus
[SUCCESS] sc.fiji/Cell_Counter
[26 minutes elapsed]
[SUCCESS] sc.fiji/Colocalisation_Analysis
[SUCCESS] sc.fiji/Color_Histogram
[SUCCESS] sc.fiji/Colour_Deconvolution
[SUCCESS] sc.fiji/CorrectBleach_
[SUCCESS] sc.fiji/Correct_3D_Drift
[SUCCESS] sc.fiji/Descriptor_based_registration
[SUCCESS] sc.fiji/Dichromacy_
[SUCCESS] sc.fiji/Directionality_
[SUCCESS] sc.fiji/FS_Align_TrakEM2
[SUCCESS] sc.fiji/Feature_Detection
[27 minutes elapsed]
[SUCCESS] sc.fiji/Fiji_Archipelago
[SUCCESS] sc.fiji/Fiji_Developer
[SUCCESS] sc.fiji/Fiji_Package_Maker
[SUCCESS] sc.fiji/Fiji_Plugins
[SUCCESS] sc.fiji/FlowJ_
[SUCCESS] sc.fiji/Graph_Cut
[SUCCESS] sc.fiji/Gray_Morphology
[SUCCESS] sc.fiji/HDF5_Vibez
[SUCCESS] sc.fiji/Helmholtz_Analysis
[SUCCESS] sc.fiji/IJ_Robot
[SUCCESS] sc.fiji/IO_
[28 minutes elapsed]
[SUCCESS] sc.fiji/Image_5D
[SUCCESS] sc.fiji/Image_Expression_Parser
[SUCCESS] sc.fiji/Interactive_3D_Surface_Plot
[SUCCESS] sc.fiji/IsoData_Classifier
[SUCCESS] sc.fiji/Kuwahara_Filter
[SUCCESS] sc.fiji/KymographBuilder
[SUCCESS] sc.fiji/LSM_Reader
[SUCCESS] sc.fiji/LSM_Toolbox
[SUCCESS] sc.fiji/Lasso_and_Blow_Tool
[SUCCESS] sc.fiji/Linear_Kuwahara
[29 minutes elapsed]
[SUCCESS] sc.fiji/LocalThickness_
[SUCCESS] sc.fiji/MTrack2_
[SUCCESS] sc.fiji/M_I_P
[SUCCESS] sc.fiji/Manual_Tracking
[SUCCESS] sc.fiji/Multi_Kymograph
[FAILURE] sc.fiji/OMEVisual
[SUCCESS] sc.fiji/PIV_analyser
[SUCCESS] sc.fiji/QuickPALM_
[SUCCESS] sc.fiji/RATS_
[SUCCESS] sc.fiji/Reconstruct_Reader
[SUCCESS] sc.fiji/SPIM_Opener
[30 minutes elapsed]
[SUCCESS] sc.fiji/SPIM_Registration
[SUCCESS] sc.fiji/Samples_
[SUCCESS] sc.fiji/Series_Labeler
[SUCCESS] sc.fiji/Simple_Neurite_Tracer
[SUCCESS] sc.fiji/Siox_Segmentation
[SUCCESS] sc.fiji/Skeletonize3D_
[SUCCESS] sc.fiji/SplineDeformationGenerator_
[SUCCESS] sc.fiji/Stack_Manipulation
[SUCCESS] sc.fiji/Statistical_Region_Merging
[31 minutes elapsed]
[SUCCESS] sc.fiji/Stitching_
[SUCCESS] sc.fiji/Sync_Win
[SUCCESS] sc.fiji/T2-NIT
[SUCCESS] sc.fiji/T2-TreelineGraph
[SUCCESS] sc.fiji/Thread_Killer
[SUCCESS] sc.fiji/Time_Lapse
[SUCCESS] sc.fiji/Time_Stamper
[SUCCESS] sc.fiji/ToAST_
[SUCCESS] sc.fiji/TopoJ_
[32 minutes elapsed]
[SUCCESS] sc.fiji/TrackMate_
[SUCCESS] sc.fiji/TrakEM2_
[SUCCESS] sc.fiji/TrakEM2_Archipelago
[SUCCESS] sc.fiji/VIB-lib
[SUCCESS] sc.fiji/VIB_
[SUCCESS] sc.fiji/Vaa3d_Reader
[SUCCESS] sc.fiji/Vaa3d_Writer
[33 minutes elapsed]
[SUCCESS] sc.fiji/VectorString
[SUCCESS] sc.fiji/Video_Editing
[SUCCESS] sc.fiji/View5D_
[SUCCESS] sc.fiji/Volume_Calculator
[SUCCESS] sc.fiji/Volume_Viewer
[SUCCESS] sc.fiji/bUnwarpJ_
[SUCCESS] sc.fiji/bigdataviewer-core
[SUCCESS] sc.fiji/bigdataviewer-server
[34 minutes elapsed]
[SUCCESS] sc.fiji/bigdataviewer-vistools
[SUCCESS] sc.fiji/bigdataviewer_fiji
[SUCCESS] sc.fiji/bigvolumeviewer
[SUCCESS] sc.fiji/bigwarp_fiji
[SUCCESS] sc.fiji/bij
[SUCCESS] sc.fiji/blockmatching_
[SUCCESS] sc.fiji/fiji
[35 minutes elapsed]
[SUCCESS] sc.fiji/fiji-lib
[SUCCESS] sc.fiji/legacy-imglib1
[SUCCESS] sc.fiji/level_sets
[SUCCESS] sc.fiji/pal-optimization
[SUCCESS] sc.fiji/panorama_
[SUCCESS] sc.fiji/register_virtual_stack_slices
[SUCCESS] sc.fiji/registration_3d
[SUCCESS] sc.fiji/spim_data
[SUCCESS] sc.fiji/trakem2-transform
[36 minutes elapsed]
[SUCCESS] sc.fiji/trakem2_tps
[SUCCESS] sc.fiji/weave_jy2java
[SUCCESS] sc.fiji/z_spacing
[SUCCESS] sc.iview/sciview
All checks succeeded! :-D
```

</details>

Problems include:

1) The build purports to pass, but the mega-melt identifies multiple failing components. Something wrong with exit codes somewhere, or the script isn't failing fast when a non-zero exit code happens?

2) imagej-ui-swing and multiview-reconstruction fail for reasons unknown (they passed for me locally at the time).

3) OMEVisual fails, probably because OpenJFX8 isn't installed. We need to add it to the `.travis.yml`, similar to how the OMEVisual project itself does:
```
before_install:
  - sudo apt-get update
  - sudo apt-get install -y openjfx libopenjfx-java libopenjfx-jni
```

4) Something in the `melting-pot.sh` script doesn't play nice on Travis, probably due to `dash`. The error happens when scanning remote tags as a fallback:
```
[WARNING] sc.fiji:Trainable_Segmentation:3.2.34: improper scm tag value; scanning remote tags…
melting-pot.sh: 428: local: Trainable_Segmentation-2.1.1: bad variable name
[ERROR] Could not fetch project source for sc.fiji:Trainable_Segmentation:3.2.34
```

--- a/.github/build.sh
+++ b/.github/build.sh
@@ -3,13 +3,19 @@ set -e
 curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-build.sh
 sh ci-build.sh
 
+# Helper method to get the last cache modified date as seconds since epoch
+last_cache_modified () {
+  cache_modified=$(find ~/.cache/scijava/melting-pot -printf '%T@\n' | sort -r | head -n1)
+  cache_modified=${cache_modified%.*}
+  echo "$cache_modified"
+}
+
 # Record the last time of cache modification before running melting-pot
 cache_modified_pre=0
 cache_modified_post=0
 
 if [ -f "~/.cache/scijava/melting-pot" ]; then
-  cache_modified_pre=$(find ~/.cache/scijava/melting-pot -printf '%T@\n' | sort -r | head -n 1)
-  cache_modified_pre=${cache_modified_pre%.*}
+  cache_modified_pre=$(last_cache_modified)
 fi
 
 # run melting-pot
@@ -17,8 +23,7 @@ tests/run.sh
 
 # Record the last time of cache modification after running melting-pot
 if [ -f "~/.cache/scijava/melting-pot" ]; then
-  cache_modified_post=$(find ~/.cache/scijava/melting-pot -printf '%T@\n' | sort -r | head -n 1)
-  cache_modified_post=${cache_modified_post%.*}
+  cache_modified_post=$(last_cache_modified)
 fi
 
 # Determine if cache needs to be re-generated

--- a/.github/build.sh
+++ b/.github/build.sh
@@ -1,4 +1,5 @@
 #!/bin/sh
+set -e
 curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-build.sh
 sh ci-build.sh
 

--- a/.github/build.sh
+++ b/.github/build.sh
@@ -1,3 +1,28 @@
 #!/bin/sh
 curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-build.sh
 sh ci-build.sh
+
+# Record the last time of cache modification before running melting-pot
+cache_modified_pre=0
+cache_modified_post=0
+
+if [ -f "~/.cache/scijava/melting-pot" ]; then
+  cache_modified_pre=$(find ~/.cache/scijava/melting-pot -printf '%T@\n' | sort -r | head -n 1)
+  cache_modified_pre=${cache_modified_pre%.*}
+fi
+
+# run melting-pot
+sh tests/run.sh
+
+# Record the last time of cache modification after running melting-pot
+if [ -f "~/.cache/scijava/melting-pot" ]; then
+  cache_modified_post=$(find ~/.cache/scijava/melting-pot -printf '%T@\n' | sort -r | head -n 1)
+  cache_modified_post=${cache_modified_post%.*}
+fi
+
+# Determine if cache needs to be re-generated
+if [ "$cache_modified_post" -gt "$cache_modified_pre" ]; then
+  echo "update-cache=true" >> $GITHUB_ENV
+else
+  echo "update-cache=false" >> $GITHUB_ENV
+fi

--- a/.github/build.sh
+++ b/.github/build.sh
@@ -13,7 +13,7 @@ if [ -f "~/.cache/scijava/melting-pot" ]; then
 fi
 
 # run melting-pot
-sh tests/run.sh
+tests/run.sh
 
 # Record the last time of cache modification after running melting-pot
 if [ -f "~/.cache/scijava/melting-pot" ]; then

--- a/.github/build.sh
+++ b/.github/build.sh
@@ -1,7 +1,6 @@
 #!/bin/sh
-set -e
-curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-build.sh
-sh ci-build.sh
+curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-build.sh &&
+sh ci-build.sh || { echo "Maven build failed. Skipping melting pot tests."; exit 1; }
 
 # Helper method to get the last cache modified date as seconds since epoch
 last_cache_modified () {
@@ -20,6 +19,7 @@ fi
 
 # run melting-pot
 tests/run.sh
+meltResult=$?
 
 # Record the last time of cache modification after running melting-pot
 if [ -f "~/.cache/scijava/melting-pot" ]; then
@@ -32,3 +32,5 @@ if [ "$cache_modified_post" -gt "$cache_modified_pre" ]; then
 else
   echo "update-cache=false" >> $GITHUB_ENV
 fi
+
+exit $meltResult

--- a/.github/setup.sh
+++ b/.github/setup.sh
@@ -1,3 +1,6 @@
 #!/bin/sh
 curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-setup-github-actions.sh
 sh ci-setup-github-actions.sh
+
+# xmllint needed for melting pot
+sudo apt-get install libxml2-utils

--- a/.github/setup.sh
+++ b/.github/setup.sh
@@ -2,5 +2,9 @@
 curl -fsLO https://raw.githubusercontent.com/scijava/scijava-scripts/master/ci-setup-github-actions.sh
 sh ci-setup-github-actions.sh
 
-# xmllint needed for melting pot
-sudo apt-get install libxml2-utils
+# Install needed packages.
+pkgs="libxml2-utils"       # needed for melting pot
+pkgs="$pkgs libxcb-shape0" # org.janelia:H5J_Loader_Plugin (fiji/H5J_Loader_Plugin@d026a1bb)
+pkgs="$pkgs libgtk2.0-0"   # net.imagej:imagej-opencv (imagej/imagej-opencv@21113e08)
+pkgs="$pkgs libblosc1"     # org.janelia.saalfeldlab:n5-blosc
+sudo apt-get -y install $(echo "$pkgs")

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -20,6 +20,7 @@ jobs:
         uses: actions/setup-java@v3
         with:
           java-version: '8'
+          java-package: 'jdk+fx'
           distribution: 'zulu'
           cache: 'maven'
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,9 +22,19 @@ jobs:
           java-version: '8'
           distribution: 'zulu'
           cache: 'maven'
+
       - name: Set up CI environment
         run: .github/setup.sh
+
+      - name: Restore the cache
+        id: restore-cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/scijava/melting-pot
+          key: ${{ runner.os }}-cache
+
       - name: Execute the build
+        id: execute-build
         run: .github/build.sh
         env:
           GPG_KEY_NAME: ${{ secrets.GPG_KEY_NAME }}
@@ -33,3 +43,19 @@ jobs:
           MAVEN_PASS: ${{ secrets.MAVEN_PASS }}
           OSSRH_PASS: ${{ secrets.OSSRH_PASS }}
           SIGNING_ASC: ${{ secrets.SIGNING_ASC }}
+
+      - name: Delete cache
+        if: env.update-cache == 'true'
+        id: delete-cache
+        run: |
+          gh api --method DELETE -H "Accept: application/vnd.github+json" /repos/scijava/pom-scijava/actions/caches?key=${{ runner.os }}-cache
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+
+      - name: Update cache post-delete
+        if: env.update-cache == 'true'
+        id: update-cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/scijava/melting-pot
+          key: ${{ runner.os }}-cache

--- a/tests/run.sh
+++ b/tests/run.sh
@@ -146,7 +146,8 @@ rm "$buildScriptTemp" ||
   die 'Error adjusting melting pot build script!'
 
 # HACK: Adjust component POMs to satisfy Maven HTTPS strictness.
-find "$meltingPotDir" -name pom.xml | while read pom
+find "$meltingPotDir" -name pom.xml |
+  while read pom
 do
   mv "$pom" "$pom.original" &&
   sed -E -e 's_(https?://maven.imagej.net|http://maven.scijava.org)_https://maven.scijava.org_g' \

--- a/tests/run.sh
+++ b/tests/run.sh
@@ -193,8 +193,47 @@ do
 done
 if [ "$doMelt" ]
 then
-  echo &&
-  cd "$meltingPotDir" && sh melt.sh || die 'Melting pot failed!'
+  echo
+  cd "$meltingPotDir"
+  sh melt.sh
+  meltResult=$?
+
+  # Dump logs for failing builds and/or tests.
+  for d in */*
+  do
+    test -d "$d" || continue
+
+    # Check for failing build log.
+    buildLog="$d/build.log"
+    if [ -f "$buildLog" ]
+    then
+      if grep -qF 'BUILD FAILURE' "$buildLog"
+      then
+        echo
+        echo "[$buildLog]"
+        cat "$buildLog"
+      fi
+    fi
+
+    # Check for failing test logs.
+    testLogsDir="$dir/target/surefire-reports"
+    if [ -d "$testLogsDir" ]
+    then
+      find "$testLogsDir" -name '*.txt' |
+        while read report
+      do
+        if grep -qF 'FAILURE!' "$report"
+        then
+          echo
+          echo "[$report]"
+          cat "$report"
+        fi
+      done
+    fi
+  done
+
+  # Terminate the script with same exit code if the melt failed.
+  test "$meltResult" -eq 0 || exit "$meltResult"
 else
   echo &&
   echo 'Melting the pot... SKIPPED'

--- a/tests/run.sh
+++ b/tests/run.sh
@@ -172,10 +172,6 @@ chmod +x "$meltScript" ||
 # Request ID: null; S3 Extended Request ID: null; Proxy: null)
 echo "org.janelia.saalfeldlab/n5-aws-s3" > "$skipTestsFile" &&
 
-# java.lang.UnsatisfiedLinkError: Unable to load library 'blosc'
-echo "org.janelia.saalfeldlab/n5-blosc" >> "$skipTestsFile" &&
-echo "org.janelia.saalfeldlab/n5-zarr" >> "$skipTestsFile" &&
-
 # Error while checking the CLIJ2 installation: null
 echo "sc.fiji/labkit-pixel-classification" >> "$skipTestsFile" ||
   die "Failed to generate $skipTestsFile"

